### ISSUE TITLE
Unnecessary specialization for fromArray, toArray

### ISSUE THREAD

**Issue by hdgarrood** [purescript-contrib/purescript-lists#16]
Since they're implemented in terms of Foldable and Unfoldable, could they be replaced by:

```purescript
fromList :: forall f a. (Unfoldable f) => List a -> f a
toList :: forall f a. (Foldable f) => f a -> List a
```

**Comment by jdegoes** [comment]
Then these tickets should go on foldable-traversable and unfoldable, right (or do they already exist?)? Then we can delete the specialized versions here.

**Comment by hdgarrood** [comment]
Neither of foldable-traversable or unfoldable currently have a dependency on lists - do you think it would be ok to add it?

Also I suppose both of these functions could potentially be replaced with just one, `foldConvert :: forall f g a. (Foldable f, Unfoldable g) => f a -> g a` (although probably with a better name)?

**Comment by hdgarrood** [comment]
Ok - sorry, my thoughts here are a little half baked. I don't think `(Foldable f, Unfoldable g)` is enough to turn an `f a` into a `g a`. I need to go away and think about this a bit more.

**Comment by paf31** [comment]
Well, we'd have a cyclic dependency if we depended on `lists`.

**Comment by paf31** [comment]
This will be fixed in the 0.7 release.
